pano/docker/pano.Dockerfile

```diff
@@ -14,16 +14,19 @@ RUN apt-get update \
     hugin \
     libvips-tools \
     python3-pip \
+    gfortran libopenblas-dev libfftw3-dev \
   && rm -rf /var/lib/apt/lists/*
 
 # pandas: pulled in as pyshtools dependency but install breaks if not mentioned explicitly (?)
 # pyshtools: used during Pannellum multires generation
 # snakemake: modern build system based on Python, manages stitching workflows
-RUN pip3 install --no-cache-dir --upgrade pip \
-  && pip3 install --no-cache-dir \
-    pandas \
-    pyshtools \
-    snakemake
+
+# Install Jupyter explicitly first
+RUN pip3 install --no-cache-dir --upgrade pip && \
+    pip3 install --no-cache-dir jupyter
+
+# Install other Python packages
+RUN pip3 install --no-cache-dir pandas pyshtools snakemake  pulp==2.7 --ignore-installed PyYAML
```

Comment:
I assume there is some reason for the pip commands to be separated, and the specific version of `pulp`, but it might be nice to explain a bit. At least to avoid somebody in the future trying to simplify it and running into exactly the same problem you did.

It's probably not a great practice generally to fix the exact version of one package and not fix the others at all, seems to be inviting future bitrot, but I guess we can live with this and fix problems when they come up.

Reply:
Yeah apparently the Snakemake has not been updated and only works with some specific package configuration due to some api changes of other libraries

Reply:
I tried having the pulp install in the same command but as weird as this sounds it had to be installed in order